src/subsystems/rust/builders/utils.nix

```diff
@@ -96,7 +97,10 @@ in rec {
       )
       dreamLock.dependencies
     );
-    lock = {inherit package;};
+    lock = utils.toTOML {inherit package;};
   in
-    utils.toTOML lock;
+    # on Nix 2.9.0pre20220527_ec07a70> builtins.toFile doesn't trigger IFD
+    if l.compareVersions l.nixVersion "2.9.0pre20220527_ec07a70" >= 0
```

Comment:
I chose this version because it was the one on my system :stuck_out_tongue: using `2.9.0` here does not work for me (as in `compareVersions` then deems my system Nix version too old, which does make sense I guess). I think it should be close enough to when reading `toFile` paths was made non-IFD, but I can get a more accurate version if it's needed.

Reply:
In general `nixVersion` branching code should be discouraged. It can lead to reproducibility issues between two different nix versions. imagine you build a cache, but some people will need different store paths because of a different nix version.
Especially in this situation where we don't need this, I would prefer to have this condition removed.
If `nixVersion` is ever used in a condition, then it should be to decide about an `abort` or `throw`, not to change the result  of the evaluation.

Reply:
Hmm yeah that makes sense; I changed it to just use `toFile`.